src/sinks/datadog/logs.rs

```diff
@@ -260,40 +275,60 @@ impl HttpSink for DatadogLogsJsonService {
                 count: events.len(),
             });
         }
-        self.config
-            .build_request(self.uri.as_str(), "application/json", body)
+        self.config.build_request(
+            self.uri.as_str(),
+            api_key.as_str(),
+            "application/json",
+            body,
+        )
     }
 }
 
 #[async_trait::async_trait]
 impl HttpSink for DatadogLogsTextService {
-    type Input = Bytes;
-    type Output = Vec<Bytes>;
+    type Input = PartitionInnerBuffer<Bytes, String>;
+    type Output = PartitionInnerBuffer<Vec<Bytes>, String>;
+
+    fn encode_event(&self, mut event: Event) -> Option<EncodedEvent<Self::Input>> {
+        let api_key = event
+            .metadata_mut()
+            .datadog_api_key
+            .take()
+            .unwrap_or_else(|| self.config.api_key.clone());
```

Comment:
You know, that's a good question. The details of what `#[inline]` _really_ means have shifted with time so I had to go look it up. The attribute is a suggestion that "the function should be inlined, including across crate boundaries". So it's a nudge to the existing inliner pass of the compiler. My preference when extracting stuff like this is to give a little nudge to make it inlined, but a preference is all that is. It may even be detrimental to runtime, as you suggest, so I retract the `#[inline]` bit of my suggestion absent evidence that a nudge is warranted.